content/en/docs/architecture/images.md

```diff
@@ -0,0 +1,132 @@
+#skip other instructions
+RUN make build --warn-undefined-variables
+
+FROM registry.ci.openshift.org/ocp/4.6:base
+COPY --from=builder /go/src/github.com/openshift/cluster-etcd-operator/bindata/bootkube/bootstrap-manifests /usr/share/bootkube/manifests/bootstrap-manifests/
+#skip other instructions
+```
+
+When the CI image is built, the images in `From` directives of `Dockerfile.rhel7` are replaced according to the `ci-operator`'s configuration above:
+* `registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.6` is replaced by `ocp_builder_rhel-8-golang-1.15-openshift-4.6`
+* `registry.ci.openshift.org/ocp/4.6:base` is replaced by `base`
+
+where `ocp_builder_rhel-8-golang-1.15-openshift-4.6` and `base` are defined in `base_images`:
+
+* `base` is `ocp/4.6:base` in the [registry of the cluster in the build farm](/docs/how-tos/use-registries-in-build-farm/) where the job is running. The image is mirrored from `registry.ci.openshift.org`. Hence, the image used in the CI build is the same as `registry.ci.openshift.org/ocp/4.6:base` in `Dockerfile.rhel7`.
+* `ocp_builder_rhel-8-golang-1.15-openshift-4.6` is `registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.6`, also the same image as in `Dockerfile.rhel7`.
+
+We use the mirrored images in the local registry instead of pulling images from `registry.ci.openshift.org` for better efficiency.
+
+## ART Images
+
+How the ART images are built is defined in the [ocp-build-data](https://github.com/openshift/ocp-build-data) repo. For example, [cluster-etcd-operator.yml](https://github.com/openshift/ocp-build-data/blob/openshift-4.6/images/cluster-etcd-operator.yml) in the `images` folder at branch `openshift-4.6` defines the build of the ART image for `cluster-etcd-operator`. 
+
+
+```yaml
+content:
+  source:
+    dockerfile: Dockerfile.rhel7
+    git:
+      branch:
+        target: release-{MAJOR}.{MINOR}
+      url: git@github.com:openshift-priv/cluster-etcd-operator.git
+from:
+  builder:
+  - stream: golang
+  member: openshift-enterprise-base
+```
+
+{{< alert title="Info" color="info" >}}
+As noticed, the github organization `openshift-priv` is used. Every commits merged to any repo of `openshift` is merged to the counterpart repo of `openshift-priv` as well. The role of `openshift-priv` is out of the scope of this page.
+{{< /alert >}}
+
+The above snippet defines
+* where to get `Dockerfile` of the build: `Dockerfile.rhel7` in `release-{MAJOR}.{MINOR}` branch of repo cloned from `git@github.com:openshift-priv/cluster-etcd-operator.git`. The branch name `release-{MAJOR}.{MINOR}` is resolved by the branch name of the `cluster-etcd-operator.yml` file. In the example, it is `release-4.6`.
+* how to replace the images in `From` directives of `Dockerfile.rhel7`: `registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.6` is replaced by `golang` and `registry.ci.openshift.org/ocp/4.6:base` is by `openshift-enterprise-base`. 
+
+## CI Images VS. ART Images
```

Comment:
I identify 3 types of images:
- CI Images (which you define)
- ART Images - images that ART builds for production releases
- ART Equivalent Images - Images that ART has made available in the CI environment so that upstream components can build atop the exact same base image packages & build images as they do downstream.

The mapping of downstream image to ART equivalent image can be found here: https://github.com/openshift/ocp-build-data/blob/openshift-4.6/streams.yml . For example, if an image uses `rhel` in its 4.6 image metadata configuration, upstream it would use: https://github.com/openshift/ocp-build-data/blob/bc10659958fc40902ffe2be6b5c86ed183aa1d25/streams.yml#L35 . This image would be called the "ART Equivalent Image" for 4.6's `rhel` stream. 

ART handles building an mirroring these images out. They are the images upstream PRs are opened to refer to.

Reply:
Updated.